[vite-plugin-angular] typechecking issue in v1.12.0

### Please provide the environment you discovered this bug in.

`"@analogjs/astro-angular": "^1.12.0",`

### Which area/package is the issue in?

vite-plugin-angular

### Description

After upgrading to the newest stable version (v1.12.0) of the @analogjs/astro-angular package, my Astro project stopped building and started throwing errors during the build process.

I think it happens because of this change in `vite-plugin-angular`:
> feat(vite-plugin-angular): fix .ts typechecking and add disableTypeChecking option by @mattlewis92 in https://github.com/analogjs/analog/pull/1538

It enables typechecking by default, which could break existing projects with different type problems.
After I set `disableTypeChecking: true` in the config, the project was successfully built.

It seems like a breaking change and maybe in the current minor version of your package - `disableTypeChecking` setting should be true by default.

### Please provide the exception or error you saw

```true
[ERROR] [vite] x Build failed in 6.06s
[@analogjs/vite-plugin-angular] [plugin @analogjs/vite-plugin-angular]
...
```


### Other information

_No response_

### I would be willing to submit a PR to fix this issue

- [ ] Yes
- [ ] No

--- a/packages/platform/src/lib/options.ts
+++ b/packages/platform/src/lib/options.ts
@@ -65,6 +65,10 @@ export interface Options {
    * requests to / in the production server build.
    */
   useAPIMiddleware?: boolean;
+  /**
+   * Disable type checking diagnostics by the Angular compiler
+   */
+  disableTypeChecking?: boolean;
 }
 
 export interface PrerenderContentDir {

--- a/packages/platform/src/lib/platform-plugin.ts
+++ b/packages/platform/src/lib/platform-plugin.ts
@@ -29,6 +29,7 @@ export function platformPlugin(opts: Options = {}): Plugin[] {
     ...angular({
       jit: platformOptions.jit,
       workspaceRoot: platformOptions.workspaceRoot,
+      disableTypeChecking: platformOptions.disableTypeChecking ?? false,
       include: [
         ...(platformOptions.include ?? []),
         ...(platformOptions.additionalPagesDirs ?? []).map(

--- a/packages/vite-plugin-angular/src/lib/angular-vite-plugin.ts
+++ b/packages/vite-plugin-angular/src/lib/angular-vite-plugin.ts
@@ -137,7 +137,7 @@ export function angular(options?: PluginOptions): Plugin[] {
     include: options?.include ?? [],
     additionalContentDirs: options?.additionalContentDirs ?? [],
     liveReload: options?.liveReload ?? false,
-    disableTypeChecking: options?.disableTypeChecking ?? false,
+    disableTypeChecking: options?.disableTypeChecking ?? true,
   };
 
   // The file emitter created during `onStart` that will be used during the build in `onLoad` callbacks for TS files

